dali/imgcodec/decoders/nvjpeg/nvjpeg_lossless.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_DECODERS_NVJPEG_NVJPEG_LOSSLESS_H_
+#define DALI_IMGCODEC_DECODERS_NVJPEG_NVJPEG_LOSSLESS_H_
+
+#include <nvjpeg.h>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/core/cuda_event.h"
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/imgcodec/decoders/decoder_batched_api_impl.h"
+#include "dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.h"
+#include "dali/pipeline/data/buffer.h"
+
+namespace dali {
+namespace imgcodec {
+
+class DLL_PUBLIC NvJpegLosslessDecoderInstance : public BatchedApiDecoderImpl {
+ public:
+  explicit NvJpegLosslessDecoderInstance(int device_id, const std::map<std::string, any> &params);
+  ~NvJpegLosslessDecoderInstance();
+
+  using ImageDecoderImpl::CanDecode;
+  bool CanDecode(DecodeContext ctx, ImageSource *in, DecodeParams opts, const ROI &roi) override;
+
+  FutureDecodeResults ScheduleDecode(DecodeContext ctx, span<SampleView<GPUBackend>> out,
+                                     cspan<ImageSource *> in, DecodeParams opts,
+                                     cspan<ROI> rois = {}) override;
+
+ private:
+  void Postprocess(DecodeResultsPromise &promise, DecodeContext ctx,
+                   span<SampleView<GPUBackend>> out, DecodeParams opts, cspan<ROI> rois);
+
+  float DynamicRangeMultiplier(int input_bpp, DALIDataType out_pixel_type) {
+    int type_bits = CHAR_BIT * dali::TypeTable::GetTypeInfo(out_pixel_type).size();
```

Comment:
```suggestion
    int type_bits = CHAR_BIT * dali::TypeTable::GetTypeInfo(out_pixel_type).size();
```
I think that it wouldn't work on any machine that doesn't have 8-bit `char` ;)
Also - signedness is missing.

I'd suggest something like:
`double DynamicRange(DALIDataType type) {
  if (!IsIntegral(type)) return 1.0;
  int positive_bits = 8 * TypeTable::GetTypeInfo(type).size() - IsSigned(type);
  return (1_u64 << positive_bits) - 1;
}
```